### PR TITLE
chore(radio-button): remove classic theme support - FE-3132

### PR DESCRIPTION
**Pull request by mkrds**
### Proposed behaviour
This PR removes classic theme support from `RadioButton` component and refactors old stories to use new CSF/MDX format.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

**Review by grabkowski — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Comment by sjohnson-sage** [comment]
This seems to have changed the text on Profile somehow if you look at the UI Tests

File: src/__experimental__/components/radio-button/radio-button.stories.mdx [hunk 1]
```diff
@@ -9,6 +9,14 @@ import { RadioButtonGroup, RadioButton } from '.';
 
 # RadioButton/RadioButtonGroup
 Used to provide a group of selectable radio buttons.
+Consider ‘smart default’ selections, based on what your user is likely to choose. But, users may well leave the defaults in place, so make sure any consequences are easy to undo, and not harmful.
+Useful to show less than about 10 options.
+Disabled or read-only radio buttons might be difficult for a user to distinguish visually, so try to avoid this.
+
+## Related Components
+- Choosing one option from a very long list? [Try Select](/?path=/docs/select--basic "Try Select").
```
Comment (DlgSHi) [inline src/__experimental__/components/radio-button/radio-button.stories.mdx:17]: the link is providing to nowhere - shouldn't it be this one instead `/?path=/docs/design-system-select--basic` ?

File: src/__experimental__/components/radio-button/radio-button.stories.mdx [hunk 2]
```diff
@@ -9,6 +9,14 @@ import { RadioButtonGroup, RadioButton } from '.';
...
+- Choosing more than one option? [Try Checkbox](/?path=/docs/experimental-checkbox--basic "Try Checkbox").
```
Comment (DlgSHi) [inline src/__experimental__/components/radio-button/radio-button.stories.mdx:18]:
```suggestion
- Choosing more than one option? [Try Checkbox](/?path=/docs/experimental-checkbox--default "Try Checkbox").
```

**Review by grabkowski — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Pull request merged**